pkg/sql/create_table.go

```diff
@@ -474,6 +475,48 @@ func resolveFK(
 		feature := fmt.Sprintf("unsupported: ON UPDATE %s", d.Actions.Update)
 		return pgerror.Unimplemented(feature, feature)
 	}
+
+	// Don't add a cascading action if there is a CHECK on any column.
+	// See #21688
+	if len(tbl.Checks) > 0 &&
+		(d.Actions.Delete == tree.Cascade ||
```

Comment:
Is it not possible to say `d.Actions.Delete != tree.NoAction` or something?
At least make a method in `tree` that checks all at once, to keep the code here shorter.

Reply:
I wanted this to be as specific and clear as possible and I did try a `!=` version and it was a lot tougher to read.  I would love to change it to a switch, but there are two variables to check so that doesn't work either.  And since this code is temporary, it seemed strange to add a helper function.